contracts/actions/common/TakeFlashloan.sol

```diff
@@ -40,7 +44,17 @@ contract TakeFlashloan is Executable, ProxyPermission {
    * @param data Encoded calldata that conforms to the FlashloanData struct
    */
   function execute(bytes calldata data, uint8[] memory) external payable override {
-    FlashloanData memory flData = parseInputs(data);
+    FlashloanData memory flDataFromProps = parseInputs(data);
+    // We need to pass the initiator address to the Operation Executor so that it can re-establish the calling context
+    FlashloanWithInitiatorData memory flData = FlashloanWithInitiatorData({
+      amount: flDataFromProps.amount,
+      asset: flDataFromProps.asset,
+      isProxyFlashloan: flDataFromProps.isProxyFlashloan,
+      isDPMProxy: flDataFromProps.isDPMProxy,
+      provider: flDataFromProps.provider,
+      calls: flDataFromProps.calls,
+      initiator: address(this)
```

Comment:
Instead of copying the data, we can pass this struct with a `zero` address as initiator from the UI and when it gets here it's gong to be updated. Any thoughts